src/core/services/theming/theming.js

```diff
@@ -7,10 +9,24 @@
 angular.module('material.core.theming', ['material.core.theming.palette'])
   .directive('mdTheme', ThemingDirective)
   .directive('mdThemable', ThemableDirective)
+  .directive('mdThemesDisabled', disableThemesDirective )
   .provider('$mdTheming', ThemingProvider)
+  .config( detectDisabledThemes )
   .run(generateAllThemes);
 
 /**
+ * Detect if the HTML or the BODY tags has a [md-themes-disabled] attribute
+ * If yes, then immediately disable all theme stylesheet generation and DOM injection
+ */
+/**
+ * @ngInject
+ */
+function detectDisabledThemes($mdThemingProvider) {
+ var isDisabled = document.querySelectorAll('[md-themes-disabled]').length > 0;
```

Comment:
Again i think we should check body and html,

We maybe can extract our `bidi` service functionality that checks `dir` attribute on `body` and `html` and make it a generic function with a callback